mobile/src/main/java/ch/epfl/sdp/mobile/application/chess/Game.kt

```diff
@@ -1,14 +1,71 @@
 package ch.epfl.sdp.mobile.application.chess
 
+import ch.epfl.sdp.mobile.application.chess.Color.Black
+import ch.epfl.sdp.mobile.application.chess.Color.White
+import ch.epfl.sdp.mobile.application.chess.Rank.*
+import ch.epfl.sdp.mobile.application.chess.implementation.BoardBuilder
+import ch.epfl.sdp.mobile.application.chess.implementation.PersistentGame
+import ch.epfl.sdp.mobile.application.chess.implementation.PersistentPieceIdentifier
+import ch.epfl.sdp.mobile.application.chess.implementation.buildBoard
+import ch.epfl.sdp.mobile.application.chess.rules.Action
+
 /**
  * An interface representing the current [Game], which contains a [Board] of pieces and on which
  * some [NextStep] should be performed to move forward for the chess game.
  */
 interface Game {
 
   /** The current [Board], which contains some pieces and should be rendered to the user. */
-  val board: Board
+  val board: Board<Piece<Color>>
 
   /** Represents the [NextStep] that must be performed on this [Game]. */
   val nextStep: NextStep
+
+  /**
+   * Returns a [Sequence] of the possible [Action] for the provided [Position]. If the [Position]
+   * does not have a [Piece] or it's not this player's turn, the resulting sequence might be empty.
+   *
+   * @param position the position for which the available [Action]s are queried.
+   * @return the [Sequence] of available actions.
+   */
+  fun actions(position: Position): Sequence<Action>
 }
+
+/** Creates a new [Game], with the standard starting positions for both players. */
+fun emptyGame(): Game =
```

Comment:
[Suggestion] Can we change it to `initializeGame`. `emtpyGame`isn't clear enough for me

Reply:
Your concern is valid, however `initializeGame` would not really follow standard Kotlin naming conventions (whereas `emptyGame` would, but I agree that it should not result in an non-empty board being created). Would a method like `Game.create()` (on the `Game` `companion object` sound good to you ?

Reply:
It's fine for me !

Reply:
Done in 65a6e465349af7486bd98dc271922d4956524656.